internal/workflow/activities/poll_sip_statuses.go

```diff
@@ -0,0 +1,116 @@
+package activities
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/google/uuid"
+	"go.artefactual.dev/tools/ref"
+	temporal_tools "go.artefactual.dev/tools/temporal"
+
+	goaingest "github.com/artefactual-sdps/enduro/internal/api/gen/ingest"
+	"github.com/artefactual-sdps/enduro/internal/enums"
+	"github.com/artefactual-sdps/enduro/internal/ingest"
+)
+
+const PollSIPStatusesActivityName = "poll-sip-statuses-activity"
+
+type PollSIPStatusesActivityParams struct {
+	BatchUUID        uuid.UUID
+	ExpectedSIPCount int
+	ExpectedStatus   enums.SIPStatus
+}
+
+type PollSIPStatusesActivityResult struct {
+	AllExpectedStatus bool
+}
+
+type PollSIPStatusesActivity struct {
+	ingestsvc    ingest.Service
+	pollInterval time.Duration
+}
```

Comment:
Suggested docstring for the activity struct:

```go
// PollSIPStatusesActivity polls the ingest service until all SIPs in a batch
// reach a final state, failing immediately if the SIP count differs from
// ExpectedSIPCount or any SIP reports an invalid final status.
```

And maybe moving this into a file-level variable would make things clearer too:

```go
// finalStatuses lists all SIP statuses that are considered terminal. Any SIP
// reporting one of these statuses is treated as final by the polling logic,
// except the caller's ExpectedStatus, which is checked separately for success.
var finalStatuses = []enums.SIPStatus{
    enums.SIPStatusFailed,
    enums.SIPStatusError,
    enums.SIPStatusCanceled,
}
```

Reply:
Done, and moved to https://github.com/artefactual-sdps/enduro/pull/1458.